src/Repository/NativeAssignmentRepository.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace App\Repository;
+
+use App\DataTransferObject\AssignmentDtoCollection;
+use App\Entity\Assignment;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\ORM\NoResultException;
+use Doctrine\ORM\ORMException;
+use Doctrine\ORM\Query\ResultSetMapping;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\Mapping\MappingException;
+
+class NativeAssignmentRepository extends AbstractRepository
+{
+    /** @var string */
+    private $kernelEnvironment;
+
+    public function __construct(ManagerRegistry $registry, string $kernelEnvironment)
+    {
+        parent::__construct($registry, Assignment::class);
+
+        $this->kernelEnvironment = $kernelEnvironment;
+    }
+
+    /**
+     * @throws ORMException
+     * @throws MappingException
+     */
+    public function insertMultiple(AssignmentDtoCollection $assignments): void
+    {
+        $queryParts = [];
+        $assignmentIndex = $this->getAvailableAssignmentStartIndex();
+
+        foreach ($assignments as $assignmentDto) {
+            $queryParts[] = sprintf(
+                "(%s, %s, %s, '%s', %d)",
+                $assignmentIndex,
+                $assignmentDto->getUserId(),
+                $assignmentDto->getLineItemId(),
+                $assignmentDto->getState(),
+                0
+            );
+
+            $assignmentIndex++;
+        }
+
+        $query = sprintf(
+            'INSERT INTO assignments (id, user_id, line_item_id, state, attempts_count) VALUES %s',
+            implode(',', $queryParts)
+        );
+
+        $this->_em->createNativeQuery($query, new ResultSetMapping())->execute();
```

Comment:
I personally would prefer it to be called entityManager, on the base class and others

Reply:
Yeah me too, this is doctrine internal stuff unfortunately :(